master/internal/sproto/task.go

```diff
@@ -255,6 +255,7 @@ type (
 		ResourcePool      string
 		Resources         ResourceList
 		JobSubmissionTime time.Time
```

Comment:
If we are going to base the calculations on ```RequestTime```, then I don't think we need JobSubmissionTime on this struct?

Reply:
Taking a look from the table task_stats perspective.

In the case of a multi-trials experiment, there will be a lot of records in the table. Some records are associated with the trials starting after the first batch. If we use JobSubmissionTime, the queued start time will be roughly the time a user creates the experiment. The start time of these records doesn't seem to match with the actual allocation start time. So, I decided to use RequestTime because the RequestTime is roughly the time an allocation starts.
  
```
"allocation_id"	"event_type"	"start_time"	"end_time"	"container_id"
"1.530f42a1-137c-43dc-a99a-08015f52afc0.1"	"QUEUED"	"2024-01-31 20:04:13.347981+00"	"2024-01-31 20:04:14.123955+00"	
"1.530f42a1-137c-43dc-a99a-08015f52afc0.1"	"QUEUED"	"0001-01-01 00:00:00+00"	"2024-01-31 20:04:43.583743+00"	
```

If we explore the other tabs [on the page](http://127.0.0.1:8080/det/resourcepool/default/stats) where we can find the "Avg Queue Time" chart, such as the ["# Active" tab](http://127.0.0.1:8080/det/resourcepool/default/active). It calls /api/v1/job-queues-v2 and lists all the jobs. Instead of using allocation_id, we could store job_id in the table and keep the event_type, start_time(JobSubmissionTime), and end_time columns. This would allow us to calculate each job's average queued time, which probably provides more meaningful information.